ddraw/IDirect3DDeviceX.cpp

```diff
@@ -1431,28 +1458,67 @@ HRESULT m_IDirect3DDeviceX::BeginScene()
 		// Set 3D Enabled
 		ddrawParent->Enable3D();
 
-		return (*d3d9Device)->BeginScene();
+		hr = (*d3d9Device)->BeginScene();
 	}
 
 	switch (ProxyDirectXVersion)
 	{
 	case 1:
-		return GetProxyInterfaceV1()->BeginScene();
+		hr = GetProxyInterfaceV1()->BeginScene();
+		break;
+
 	case 2:
-		return GetProxyInterfaceV2()->BeginScene();
+		hr = GetProxyInterfaceV2()->BeginScene();
+		break;
+
 	case 3:
-		return GetProxyInterfaceV3()->BeginScene();
+		hr = GetProxyInterfaceV3()->BeginScene();
+		break;
+
 	case 7:
-		return GetProxyInterfaceV7()->BeginScene();
-	default:
-		return DDERR_GENERIC;
+		hr = GetProxyInterfaceV7()->BeginScene();
+		break;
+	}
+
+#if WITH_IMGUI
+	if(SUCCEEDED(hr) || Config.Dd7to9)
+	{
+		ImGui_ImplDX9_NewFrame();
+		ImGui_ImplWin32_NewFrame();
+		ImGui::NewFrame();
 	}
+#endif
```

Comment:
Putting the ImGui outside of `Config.Dd7to9` causes a crash if `EnableDdrawWrapper` is enabled but not `Dd7to9`.  We should probably only run ImGui when `Config.Dd7to9` is enabled, since I don't know if ImGui can support DirectDraw or Direct3D7.